std/protocols/lookup.asm

```diff
@@ -51,21 +54,29 @@ let compute_next_z: Fp2<expr>, Fp2<expr>, Fp2<expr>, Constr, expr -> fe[] = quer
     unpack_ext_array(res)
 };
     
-/// Adds constraints that enforce that rhs is the lookup for lhs
+/// Transfroms a single lookup constraint to identity constraint, challenges and
+/// higher-stage witness columns.
+/// Use this function if the backend does not support lookup constraints natively.
+/// WARNING: This function can currently not be used multiple times since
+/// the used challenges would overlap.
+/// TODO: Implement this for an array of constraints.
 /// Arguments:
-/// - acc: A phase-2 witness column to be used as the accumulator. If 2 are provided, computations
-///        are done on the F_{p^2} extension field.
-/// - alpha: A challenge used to compress the LHS and RHS values
-/// - beta: A challenge used to update the accumulator
 /// - lookup_constraint: The lookup constraint
-/// - multiplicities: The multiplicities which shows how many times each RHS value appears in the LHS                  
-let lookup: expr[], Fp2<expr>, Fp2<expr>, Constr, expr -> () = constr |acc, alpha, beta, lookup_constraint, multiplicities| {
+/// - multiplicities: A multiplicities column which shows how many times each row of the RHS value appears in the LHS                  
+let lookup: Constr, expr -> () = constr |lookup_constraint, multiplicities| {
+    std::check::assert(required_extension_size() <= 2, || "Invalid extension size");
```

Comment:
Could this be removed if below we use `2*i` and `2*i + 1` as challenge ids?

Reply:
The IDs were the main reason for this assert, but also we cannot use fp2 for larger arrays.